app/_src/guides/builtin-gateway.md

```diff
@@ -0,0 +1,340 @@
+a builtin gateway.
+
+In the [quickstart](/docs/{{ page.version }}/quickstart/kubernetes-demo/), traffic was only able to get in the mesh by port-forwarding to an instance of an app
+inside the mesh.
+This is not something that can be used in production and you need to set up a gateway to receive traffic external to the mesh.
+In this guide you will add a gateway in front of the demo-app service to expose it publicly.
+
+{% mermaid %}
+---
+title: service graph of the demo app with a builtin gateway on front
+---
+flowchart LR
+  subgraph edge-gateway
+    gw0(/ :8080)
+  end
+  demo-app(demo-app :5000)
+  redis(redis :6379)
+  gw0 --> demo-app 
+  demo-app --> redis
+{% endmermaid %}
+
+## Prerequisites
+- Completed [quickstart](/docs/{{ page.version }}/quickstart/kubernetes-demo/) to set up a zone control plane with demo application
+
+## Start a gateway 
+
+### Create a `MeshGatewayInstance` 
+
+A [`MeshGatewayInstance`](/docs/{{ page.version }}/using-mesh/managing-ingress-traffic/builtin-k8s/) lets you configure the pods
+that will run your gateway.
+
+You can do so by running:
+
+```shell
+echo "
+apiVersion: kuma.io/v1alpha1
+kind: MeshGatewayInstance
+metadata:
+  name: edge-gateway
+  namespace: kuma-demo
+spec:
+  replicas: 2
+  serviceType: LoadBalancer
+  tags:
+    kuma.io/service: edge-gateway
+" | kubectl apply -f -
+```
+
+{% warning %}
+Your kubernetes cluster needs to support LoadBalancer for this to work.
+This may not be the case if you are running kubernetes locally with `kind` or `k3d`. 
+{% endwarning %}
+
+### Define a listener using `MeshGateway`
+
+[`MeshGateway`](/docs/{{ page.version }}/using-mesh/managing-ingress-traffic/builtin-listeners/) lets you define listeners for your gateway.
+
+Here we will define a single HTTP listener on port 8080:
+
+```shell
+echo "
+apiVersion: kuma.io/v1alpha1
+kind: MeshGateway
+mesh: default
+metadata:
+  name: my-gateway
+spec:
+  selectors:
+    - match:
+        kuma.io/service: edge-gateway
+  conf:
+    listeners:
+      - port: 8080
+        protocol: HTTP
+        tags:
+          port: http/8080
+" | kubectl apply -f -
+```
+
+Notice how the selector selects the `kuma.io/service` tag of the previously defined `MeshGatewayInstance`.
+
+Now that you have this running you can see gateway pods running:
+
+```shell
+$ kubectl get pods -n kuma-demo
+NAME                            READY   STATUS    RESTARTS   AGE
+redis-5fdb98848c-5tw62          2/2     Running   0          5m5s
+demo-app-c7cd6588b-rtwlj        2/2     Running   0          5m5s
+edge-gateway-66c76fd477-ncsp5   1/1     Running   0          18s
+edge-gateway-66c76fd477-thxqj   1/1     Running   0          18s
+```
+
+You can then retrieve the public url for your gateway with:
+
+```shell
+export PROXY_IP=$(kubectl get svc --namespace kuma-demo edge-gateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+echo $PROXY_IP
+```
```

Comment:
So this is similar to what we were discussing offline. If I follow the quickstart using `kind` this command doesn't print anything for me. If I use just `.status` I see:

```
kubectl get svc --namespace kuma-demo edge-gateway -o jsonpath='{.status}'

{"loadBalancer":{}}
```

BTW, I tried with `minikube` this morning after our catchup and I did manage to get it forwarding to the gateway with `minikube service` - not sure what the alternative would be with `kind`?

Reply:
Yes I think I'm going to update both guides to use minkube instead. It's a bit nicer.

However, I'll do this in a follow up PR.

Reply:
Gotcha sounds good, the rest of the walkthrough worked fine for me, will resolve this one.

Reply:
opened: https://github.com/kumahq/kuma-website/issues/1649